casework/common.py

```
@@ -736,6 +737,173 @@ def parse_extraction_response(response_text, wrapper_keys):
+  as given to you, e.g. "… (080-CR-0047)". Keep the hook from crowding it out.
+- Keep it under ~160 characters."""
+
+# A court case number like 080-CR-0047 / 081-WO-1234. Case-insensitive to match
+# the review gate's detector; the lookbehind/ahead anchor the token so a
+# malformed adjacent token can't match the real number as a substring.
+COURT_RE = re.compile(r"(?<![\dA-Za-z])\d{2,3}-[A-Za-z]{1,3}-\d{3,4}(?![\dA-Za-z])")
+
+# Defendant-headcount patterns a title must avoid: a Devanagari/ASCII number
+# immediately followed by जना / व्यक्ति / प्रतिवादी. The court number itself
+# (080-CR-0098) won't match — no such trailing noun.
+_HEADCOUNT_RE = re.compile(r"[०-९0-9]+\s*(जना|व्यक्ति|प्रतिवादी)")
```

Ruff (0.15.17)

[warning] String contains ambiguous `०` (DEVANAGARI DIGIT ZERO). Did you mean `o` (LATIN SMALL LETTER O)?

(RUF001)